index.js

```diff
@@ -1,14 +1,29 @@
 'use strict';
+
 require('harmony-reflect');
 
-module.exports = function (obj) {
-	return new Proxy(obj, {
+module.exports = function (target, readonly) {
+	var handler = {
 		get: function (target, name, receiver) {
 			if (!(name in target)) {
 				throw new TypeError('Unknown property: ' + name);
 			}
 
 			return Reflect.get(target, name, receiver);
+		},
+
+		set: function(target, name, val, receiver) {
+			if (!(name in target) && readonly) {
+				throw new Error('Readonly object: Please don\'t set properties on this object.');
+			}
+
+			target[name] = val;
```

Comment:
I think we must use Reflect.set here, see https://github.com/tvcutsem/harmony-reflect/blob/master/doc/api.md#reflectsettarget-name-value-receiver